src/Wallet.ts

```diff
@@ -0,0 +1,252 @@
+import * as bip39 from 'bip39';
+import * as bip32 from 'bip32';
+import * as bitcoinjs from 'bitcoinjs-lib';
+import coinSelectAccumulative, { Target } from 'coinselect/accumulative';
+import { BlockchainService, Utxo } from './BlockchainService';
+
+export type AddressMetadata = {
+  address: string;
+  derivationPath: string;
+  publicKey: Buffer;
+};
+
+export type WalletStats = {
+  balance: number;
+  pendingBalance: number;
+  nextUnusedAddress: AddressMetadata;
+  nextUnusedChangeAddress: AddressMetadata;
+  addresses: AddressMetadata[];
+  changeAddresses: AddressMetadata[];
+  addressesWithUtxo: AddressMetadata[];
+};
+
+export type UtxoWithMetadata = Utxo & {
+  address: string;
+  derivationPath: string;
+};
+
+export type AddressCollectionStats = Omit<
+  WalletStats,
+  'nextUnusedChangeAddress' | 'changeAddresses'
+>;
+
+export class Wallet {
```

Comment:
Below is the Wallet that a developer would look at (along with the included tests) to learn how to build a BitcoinJS Wallet or even use it directly in a production environment...